src/generators/python.ts

```diff
@@ -0,0 +1,258 @@
+        }
+        return makeClassName(fhirType);
+    }
+
+    curlyBlock(tokens: string[], gencontent: () => void) {
+        this.write(tokens.join(' '));
+        this.write(':\n');
+        this.ident();
+        gencontent();
+        this.deident();
+        this.write('\n');
+    }
+
+    wrapOptional(s: string) {
+        return 'Optional[' + s + ']';
+    }
+
+    wrapList(s: string) {
+        return 'List[' + s + ']';
+    }
+
+    generateType(schema: TypeSchema) {
+        const className = makeClassName(schema.name);
+        const superClasses = [...(schema.base ? [schema.base.name] : []), ...injectSuperClasses(schema.name.name)];
+        const classDefinition = `class ${className}(${superClasses.join(", ")})`;
+
+        this.curlyBlock([classDefinition], () => {
+            if (!schema.fields) {
+                this.line('pass');
+                return;
+            }
+
+            let fields = Object.entries(schema.fields).sort((a, b) => a[0].localeCompare(b[0]));
+            this.line(`resourceType: Literal['${className}'] = '${className}'`)
+
+            for (const [fieldName, field] of fields) {
+                let fieldType = this.toLangType(field.type);
+                let defaultValue = '';
+
+                if (field.array) {
+                    fieldType = this.wrapList(fieldType);
+                }
+
+                if (!field.required) {
+                    fieldType = this.wrapOptional(fieldType);
+                    defaultValue = ' = None';
+                }
+
+                const fieldDecl = `${fixReservedWords(snakeCase(fieldName))}: ${fieldType}${defaultValue}`;
+                this.line(fieldDecl);
+            }
+        });
+    }
+
+    default_imports() {
+        this.line('from __future__ import annotations');
+        this.line('from', 'pydantic', 'import', '*');
+        this.line('from', 'typing', 'import', ['Optional', 'List', 'Literal'].join(', '));
+    }
+
+    generateNestedTypes(schema: TypeSchema) {
+        if (schema.nestedTypes) {
+            this.line('# Nested Types');
+            this.line();
+            for (let subtype of schema.nestedTypes) {
+                this.generateType(subtype);
+            }
+        }
+    }
+
+    generateDependenciesImports(schema: TypeSchema) {
+        if (schema.allDependencies) {
+            for (const deps of schema.allDependencies) {
+                if (deps.type === 'resource') {
+                    this.line('from', `.${snakeCase(deps.name)}`, 'import', '*');
+                }
+            }
+        }
+    }
+
+    generateBaseModel() {
+        this.curlyBlock(['class BaseModel_(BaseModel)'], async () => {
+            this.line('pass');
+        })
+    }
+
+    generateReExport() {
+
+    }
+
+    generate() {
+        this.dir('.', async () => {
+            this.file('__init__.py', () => { });
+            const groupedComplexTypes = groupedByPackage(this.loader.complexTypes())
```

Comment:
Maybe a little bit refactoring this part to make it more similar to output structure? After time hard to read.

Reply:
I am not quite sure what do you mean by "output structure". 
 Could you please elaborate, perhaps with an example?

Reply:
I mean output project structure. When I read the code it will be quite tricky to separate base resources file and all other files. But maybe -- it will be more simple to show what I mean. Will send commit soon.

Reply:
https://github.com/fhir-schema/fhir-schema-codegen/pull/4/commits/0b732ef01b42ae97f068b1961e6d6189d4769da0